plugins/org.eclipse.fordiac.ide.globalconstantseditor.ui/src/org/eclipse/fordiac/ide/globalconstantseditor/ui/handlers/GlobalConstantsFindReferencesHandler.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Daniel Lindhuber
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.globalconstantseditor.ui.handlers;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.model.search.ModelQuerySpec;
+import org.eclipse.fordiac.ide.model.search.ModelQuerySpec.SearchScope;
+import org.eclipse.fordiac.ide.model.search.ModelSearchQuery;
+import org.eclipse.fordiac.ide.structuredtextcore.stcore.STVarDeclaration;
+import org.eclipse.search.ui.NewSearchUI;
+import org.eclipse.xtext.ui.editor.findrefs.FindReferencesHandler;
+
+@SuppressWarnings("restriction")
+public class GlobalConstantsFindReferencesHandler extends FindReferencesHandler {
+
+	@Override
+	protected void findReferences(final EObject target) {
+		if (target instanceof final STVarDeclaration varDec) {
```

Comment:
Please check that it is a global constant and not a local variable with the additional condition:
```suggestion
		if (target instanceof final STVarDeclaration varDec && varDec.eContainer() instanceof STVarGlobalDeclarationBlock) {
```